test/acceptance/CustomsNotificationRetryFailureSpec.scala

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package acceptance
+
+import org.scalatest.{Matchers, OptionValues}
+import play.api.Application
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.mvc._
+import play.api.test.Helpers._
+import uk.gov.hmrc.customs.notification.repo.NotificationWorkItemMongoRepo
+import uk.gov.hmrc.mongo.MongoSpecSupport
+import uk.gov.hmrc.workitem._
+import util.TestData._
+import util._
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class CustomsNotificationRetryFailureSpec extends AcceptanceTestSpec
+  with Matchers
+  with OptionValues
+  with ApiSubscriptionFieldsService
+  with NotificationQueueService
+  with PushNotificationService
+  with MongoSpecSupport {
+
+  private val endpoint = "/customs-notification/notify-retry"
+
+  private lazy val repo = app.injector.instanceOf[NotificationWorkItemMongoRepo]
+
+
+  private lazy val pollerConfigs = Map(
+    "push.retry.initialPollingInterval.milliseconds" -> 100,
+    "push.retry.retryAfterFailureInterval.seconds" -> 1,
+    "push.retry.inProgressRetryAfter.seconds" -> 1,
+    "push.retry.poller.instances" -> 10,
+    "unblock.polling.delay.duration.milliseconds" -> 100
+  )
+
+  override implicit lazy val app: Application = new GuiceApplicationBuilder().configure(acceptanceTestConfigs ++ pollerConfigs).build()
+
+  override protected def beforeEach() {
+    await(repo.drop)
+    startMockServer()
+  }
+
+  override protected def afterEach(): Unit = {
+    resetMockServer()
+    stopMockServer()
+    await(repo.drop)
+  }
+
+  feature("Ensure offline retry") {
+    scenario("backend submits a valid PUSH request with incorrect callback details used resulting in endpoint returning a 404") {
+
+      startApiSubscriptionFieldsService(validFieldsId,callbackData)
+      Given("the Push endpoint is setup to return NOT_FOUND")
+      setupPushNotificationServiceToReturn(NOT_FOUND)
```

Comment:
test passes when this line is removed, like below. Is there something wrong with these acceptance tests?

Reply:
`setupPushNotificationServiceToReturn(NOT_FOUND)` will program a failure response. Not having this line will have the same effect as this means the endpoint does not respond successfully.